sklearn/utils/estimator_checks.py

```diff
@@ -3293,6 +3294,7 @@ def check_non_transformer_estimators_n_iter(name, estimator_orig):
         "LinearSVC",
         "LogisticRegression",
         "SelfTrainingClassifier",
+        "BisectingKMeans",
```

Comment:
I think the issue is before this point. A transformer should not end up here since it's explicitely stated that this is a test for estimators that are not transformers. I think that we should not yield this check in ``_yield_clustering_checks`` if it's a transformer.